kolibri/plugins/learn/assets/src/views/classes/Hint.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+
+  <div class="hint-btn-container">
+    <KButton
+      class="hint-btn"
+      appearance="basic-link"
+      :text="$tr('hint', { hintsLeft: availableHints })"
+      :primary="false"
+      @click="takeHint"
```

Comment:
I don't see a method called `takeHint` defined anywhere - is something missing here? I tested a hint and it worked alright but not sure how

Reply:
I am also somewhat surprised by this, as `takeHint` is not defined on the `KContentRenderer` functional component either: https://github.com/learningequality/kolibri-design-system/blob/main/lib/content/KContentRenderer.js#L10

Reply:
Wondering if you all were using the hint in the top bar (where NaN is here) or the one on the bottom (1 left)? The one on the top is still WIP as part of the refactor that @sairina  is working on. The one at the bottom is the current implementation, which hasn't been taken out yet as part of that refactor. I _think_ what's happening is the one in the renderer is working, and "takeHint" is for the top bar, and isn't defined/working, because Sairina is still working on it
<img width="855" alt="Screen Shot 2021-08-25 at 9 33 53 AM" src="https://user-images.githubusercontent.com/17235236/130800108-14520fab-b9b6-4369-a0d3-258972f8f787.png">

Reply:
For context, I did not test this - I was just surprised at the report.

One issue with this hint refactor out of the Perseus renderer and into the top bar is that it assumes certain things about hints that are specific to Perseus - QTI also supports hints but has a slightly different mechanism as far as I can tell.

I wonder if instead of directly encoding this here, we should be exposing a slot that an exercise renderer can inject its 'hint prompting' into (this has parallels with suggestions elsewhere to expose a common 'toolbar' that renderers inject their control components into like the EPUB renderer).